### PR TITLE
Feature/add op test

### PR DESCRIPTION
**Pull request by reyoung**
Review #2945 First.

### CODE REVIEW

File: python/paddle/v2/framework/tests/test_add_two_op.py [hunk 1]
```diff
@@ -0,0 +1,17 @@
+import unittest
+from op_test_util import OpTestMeta
+import numpy
```
Comment (Superjom) [inline python/paddle/v2/framework/tests/test_add_two_op.py]: `import numpy as np`

File: paddle/pybind/pybind.cc [hunk 2]
```diff
@@ -14,6 +14,7 @@ limitations under the License. */
 
 #include <Python.h>
 #include <paddle/framework/op_registry.h>
```
Comment (Superjom) [inline paddle/pybind/pybind.cc]:
```c++
#include "paddle/framework/op_registry.h"
```

others same.

File: paddle/framework/net.cc [hunk 3]
```diff
@@ -19,7 +19,10 @@
 namespace paddle {
 namespace framework {
 
-void PlainNet::CompleteAddOp() {
+void PlainNet::CompleteAddOp(bool calc) {
```
Comment (jacquesqiao) [inline paddle/framework/net.cc]: 为什么需要 calc？

Comment (reyoung) [inline paddle/framework/net.cc]: 因为其实继承产生的Net，是不需要重新计算哪些输入和输出的。

File: python/paddle/v2/framework/tests/test_add_two_op.py (hunk 1 above)
Comment (reyoung) [inline python/paddle/v2/framework/tests/test_add_two_op.py]: I think it doesn't matter.

File: paddle/pybind/pybind.cc (hunk 2 above)
Comment (reyoung) [inline paddle/pybind/pybind.cc]: I will give another PR fix them all.

File: paddle/framework/attr_checker.h [hunk 4]
```diff
@@ -41,6 +42,35 @@ class DefaultValueSetter {
   T default_value_;
 };
 
+template <typename T>
+class EnumInContainer {
```
Comment (jacquesqiao) [inline paddle/framework/attr_checker.h]: this class is used as checker, so if EnumInContainerChecker a better name?

File: paddle/operators/fc_op.cc [hunk 5]
```diff
@@ -0,0 +1,76 @@
...
+    AddOp(framework::OpRegistry::CreateOp("mul",
+                                          {
+                                              Input("X"), Input("W"),
+                                          },
+                                          {Output("before_act")},
+                                          {}));
+    auto b = Input("b");
+    if (b != framework::OperatorBase::EMPTY_VAR_NAME()) {
+      AddOp(framework::OpRegistry::CreateOp("rowwise_add",
+                                            {Output("before_act"), Input("b")},
```
Comment (jacquesqiao) [inline paddle/operators/fc_op.cc]: If `act` is a good Abbreviation for activation?

File: python/paddle/v2/framework/tests/test_fc_op.py [hunk 6]
```diff
@@ -0,0 +1,43 @@
+import paddle.v2.framework.core as core
```
Comment (jacquesqiao) [inline python/paddle/v2/framework/tests/test_fc_op.py]: 这个test为何不用 OpTestMeta？

File: python/paddle/v2/framework/tests/op_test_util.py [hunk 7]
```diff
@@ -0,0 +1,50 @@
+import paddle.v2.framework.core as core
+import unittest
+import numpy
+import paddle.v2.framework.create_op_creation_methods as creation
+
+
+class OpTestMeta(type):
```
Comment (jacquesqiao) [inline python/paddle/v2/framework/tests/op_test_util.py]: can you explain the Principle of this meta class?

File: python/paddle/v2/framework/create_op_creation_methods.py [hunk 8]
```diff
@@ -217,6 +217,10 @@ def __impl__(*args, **kwargs):
         return core.Operator.create(opdesc.SerializeToString())
 
     __impl__.__doc__ = get_docstring_from_op_proto(op_proto)
+    __impl__.all_input_args = [var.name for var in op_proto.inputs]
+    __impl__.all_output_args = [var.name for var in op_proto.outputs]
```
Comment (jacquesqiao) [inline python/paddle/v2/framework/create_op_creation_methods.py]: I think following is better
```shell
all_input_names
all_output_names
all_attrs_names
```

Comment (reyoung) [inline python/paddle/v2/framework/create_op_creation_methods.py]: It is not `input_name`, it is `input argument name` of that Op function.

File: paddle/framework/attr_checker.h (hunk 4 above)
Comment (reyoung) [inline paddle/framework/attr_checker.h]: If that names are really matter, I prefer add a sub namespace to do this.
```cpp
namespace paddle {
namespace framework {
namespace attr_checkers {
}
}
}
```

File: python/paddle/v2/framework/tests/test_fc_op.py (hunk 6 above)
Comment (reyoung) [inline python/paddle/v2/framework/tests/test_fc_op.py]: Because there are many op not implemented yet.

**Comment by reyoung** [comment]
I merge this PR first because many other PRs depends on this PR.  I will add comments to `OpTestMeta` in another PR.

**Pull request merged**